File: win-prog/ProgWinCS/Files and Streams/HtmlDump/HtmlDump.cs
//---------------------------------------
// HtmlDump.cs © 2001 by Charles Petzold
//---------------------------------------
using System;
using System.IO;
using System.Net;

class HtmlDump
{
     public static int Main(string[] astrArgs)
     {
          if (astrArgs.Length == 0)
          {
               Console.WriteLine("Syntax: HtmlDump URI");
               return 1;
          }

          WebRequest webreq;
          WebResponse webres;

          try
          {
               webreq = WebRequest.Create(astrArgs[0]);
               webres = webreq.GetResponse();
          }
          catch (Exception exc)
          {
               Console.WriteLine("HtmlDump: {0}", exc.Message);
               return 1;
          }

          if (webres.ContentType.Substring(0, 4) != "text")
          {
               Console.WriteLine("HtmlDump: URI must be a text type.");
               return 1;
          }

          Stream       stream = webres.GetResponseStream();
          StreamReader strrdr = new StreamReader(stream);
          string       strLine;
          int          iLine = 1;

          while ((strLine = strrdr.ReadLine()) != null)
               Console.WriteLine("{0:D5}: {1}", iLine++, strLine);

          stream.Close();
          return 0;
     }
}

File: win-prog/ProgWinVB/Files and Streams/HtmlDump/HtmlDump.vb
'-----------------------------------------
' HtmlDump.vb (c) 2002 by Charles Petzold
'-----------------------------------------
Imports System
Imports System.IO
Imports System.Net

Module HtmlDump
    Function Main(ByVal astrArgs() As String) As Integer
        Dim strUri As String

        ' If there's no argument, display syntax and ask for URI.
        If astrArgs.Length = 0 Then
            Console.WriteLine("Syntax: HtmlDump URI")
            Console.Write("Enter URI: ")
            strUri = Console.ReadLine()
        Else
            strUri = astrArgs(0)
        End If

        ' If the user enters a blank string, dive out.
        If strUri.Length = 0 Then
            Return 1
        End If

        Dim webreq As WebRequest
        Dim webres As WebResponse

        Try
            webreq = WebRequest.Create(strUri)
            webres = webreq.GetResponse()
        Catch exc As Exception
            Console.WriteLine("HtmlDump: {0}", exc.Message)
            Return 1
        End Try

        If webres.ContentType.Substring(0, 4) <> "text" Then
            Console.WriteLine("HtmlDump: URI must be a text type.")
            Return 1
        End If

        Dim strm As Stream = webres.GetResponseStream()
        Dim sr As New StreamReader(strm)
        Dim strLine As String
        Dim iLine As Integer = 1

        strLine = sr.ReadLine()
        While Not strLine Is Nothing
            Console.WriteLine("{0:D5}: {1}", iLine, strLine)
            iLine += 1
            strLine = sr.ReadLine()
        End While

        strm.Close()
        Return 0
    End Function
End Module
